graylog2-web-interface/src/components/common/PaginatedList.jsx

```diff
@@ -46,9 +46,14 @@ class PaginatedList extends React.Component {
   };
 
   componentWillReceiveProps(nextProps) {
-    if (this.props.activePage !== nextProps.activePage) {
+    const { pageSize, activePage } = this.props;
+
+    if (activePage !== nextProps.activePage) {
       this.setState({ currentPage: nextProps.activePage });
     }
+    if (pageSize !== nextProps.pageSize) {
+      this.setState({ pageSize: nextProps.pageSize });
```

Comment:
I was wondering if calling `setState()` twice in this method will trigger a double re-render of components?
If that's the case, can we only do one `setState()`?

Reply:
No, calling `setState()` only enqueues the changes but the re-render does not happen immediately.

From https://reactjs.org/docs/react-component.html#setstate:
> Think of `setState()` as a request rather than an immediate command to update the component. For better perceived performance, React may delay it, and then update several components in a single pass. React does not guarantee that the state changes are applied immediately.

Reply:
Okay, got it. Thanks for the explanation. :+1: